packages/browser/src/utils/lazyLoadIntegration.ts

```diff
@@ -0,0 +1,90 @@
+import { SDK_VERSION } from '@sentry/core';
+import type { IntegrationFn } from '@sentry/types';
+import { WINDOW } from '../helpers';
+
+// This is a map of integration function method to bundle file name.
+const LazyLoadableIntegrations = {
+  replayIntegration: 'replay',
+  replayCanvasIntegration: 'replay-canvas',
+  feedbackIntegration: 'feedback',
+  captureConsoleIntegration: 'captureconsole',
+  contextLinesIntegration: 'contextlines',
+  linkedErrorsIntegration: 'linkederrors',
+  debugIntegration: 'debug',
+  dedupeIntegration: 'dedupe',
+  extraErrorDataIntegration: 'extraerrordata',
+  httpClientIntegration: 'httpclient',
+  reportingObserverIntegration: 'reportingobserver',
+  rewriteFramesIntegration: 'rewriteframes',
+  sessionTimingIntegration: 'sessiontiming',
+} as const;
+
+const WindowWithMaybeIntegration = WINDOW as typeof WINDOW & {
+  Sentry?: Partial<Record<keyof typeof LazyLoadableIntegrations, IntegrationFn>>;
+};
+
+/**
+ * Lazy load an integration from the CDN.
+ * Rejects if the integration cannot be loaded.
+ */
+export async function lazyLoadIntegration(name: keyof typeof LazyLoadableIntegrations): Promise<IntegrationFn> {
+  const bundle = LazyLoadableIntegrations[name];
+
+  if (!bundle || !WindowWithMaybeIntegration.Sentry) {
+    throw new Error(`Cannot lazy load integration: ${name}`);
+  }
+
+  // Bail if the integration already exists
+  const existing = WindowWithMaybeIntegration.Sentry[name];
+  console.log({ existing });
+  if (typeof existing === 'function') {
+    return existing;
```

Comment:
If it exists, does it *only* mean that the integration already exists in their main SDK bundle? Or does the integration get attached to the Sentry namespace if we load it (e.g. via CDN below)

Reply:
this should only exist for CDN bundles - the NPM package will not attach this to the global, so it would not capture this! (which is what we want here I believe)